core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -34,7 +40,7 @@ const getMarketPrice = async function (
     collateralAmount: BigNumber
 ): Promise<BigNumber> {
     // convert collateral into DAI
-    const daiAmount = await convertCollateralToDaiUsingRoute(network, collateral.symbol, marketId, collateralAmount);
+    const daiAmount = await convertCollateralToDaiUsingPool(network, collateral.symbol, marketId, collateralAmount);
```

Comment:
`convertCollateralToDaiUsingPool` means you always generate a new pool there (as it's not provided as an optional argument)?

Reply:
yes, also in the function body the route is always converted to pool